deploy/kubernetes_operator/pkg/operator/app_manager.go

```diff
@@ -584,6 +588,10 @@ func (am *appManager) isAppTimeOut(app *v1alpha1.FLApp, now time.Time) bool {
 func (am *appManager) syncFailingApp(ctx context.Context, app *v1alpha1.FLApp) error {
 	name := app.Name
 	klog.Infof("sync failing app, name = %v", name)
+
+	// Update flapp replica status to failed.
+	updateReplicaStatus(app, v1alpha1.FLStateFailed)
```

Comment:
This may not work because it only update app's status in memory, however we want to persist it in CRD.

Reply:
Besides, I think this should be in `syncShuttingDownApp`. The state machine transit from `failing` to `shutingdown`